make/tools.mk

```diff
@@ -401,17 +397,25 @@ $(BINDIR)/downloaded/tools/kube-apiserver-kubebuilder-$(KUBEBUILDER_ASSETS_VERSI
 	tar xfO $< kubebuilder/bin/kube-apiserver > $@ && chmod 775 $@
 
 $(BINDIR)/downloaded/tools/kubebuilder-tools-$(KUBEBUILDER_ASSETS_VERSION)-$(HOST_OS)-$(HOST_ARCH).tar.gz: | $(BINDIR)/downloaded/tools
-ifeq ($(HOST_OS)-$(HOST_ARCH),darwin-arm64)
-	@$(eval KUBEBUILDER_ARCH := amd64)
-	$(warning Downloading amd64 kubebuilder-tools for integration tests since darwin/arm64 isn't currently packaged. This will require rosetta in order to work)
-else
-	@$(eval KUBEBUILDER_ARCH := $(HOST_ARCH))
-endif
-	curl -sSfL https://storage.googleapis.com/kubebuilder-tools/kubebuilder-tools-$(KUBEBUILDER_ASSETS_VERSION)-$(HOST_OS)-$(KUBEBUILDER_ARCH).tar.gz > $@
+	curl -sSfL https://storage.googleapis.com/kubebuilder-tools/kubebuilder-tools-$(KUBEBUILDER_ASSETS_VERSION)-$(HOST_OS)-$(HOST_ARCH).tar.gz > $@
+
+##############
+# gatewayapi #
+##############
+
+GATEWAY_API_SHA256SUM=0eed80ad85850a6cbd17ab705aea59e49641c7bf1e6d3fbed1fc0156ffd62734
 
-$(BINDIR)/downloaded/gatewayapi-v%: | $(BINDIR)/downloaded
+$(BINDIR)/downloaded/gatewayapi-v$(GATEWAY_API_VERSION): $(BINDIR)/downloaded/gatewayapi-v$(GATEWAY_API_VERSION).tar.gz | $(BINDIR)/downloaded
+	./hack/util/checkhash.sh $< $(GATEWAY_API_SHA256SUM)
 	@mkdir -p $@
-	curl -sSfL https://github.com/kubernetes-sigs/gateway-api/archive/refs/tags/v$*.tar.gz | tar xz -C $@
+	tar xz -C $@ -f $<
+
+$(BINDIR)/downloaded/gatewayapi-v$(GATEWAY_API_VERSION).tar.gz: | $(BINDIR)/downloaded
+	curl -sSfL https://github.com/kubernetes-sigs/gateway-api/archive/refs/tags/v$(GATEWAY_API_VERSION).tar.gz > $@
```

Comment:
note: I spotted this while updating the kubebuilder stuff; we generally don't use anything else without checking its hash first so I figured that such a check should be added here!